compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -290,14 +303,14 @@ class TreeUnpickler(reader: TastyReader,
         val end = readEnd()
 
         def readMethodic[N <: Name, PInfo <: Type, LT <: LambdaType]
-            (companion: LambdaTypeCompanion[N, PInfo, LT], nameMap: Name => N): LT = {
+            (companionOp: FlagSet => LambdaTypeCompanion[N, PInfo, LT], nameMap: Name => N): LT = {
           val result = typeAtAddr.getOrElse(start, {
               val nameReader = fork
               nameReader.skipTree() // skip result
               val paramReader = nameReader.fork
-              val paramNames = nameReader.readParamNames(end).map(nameMap)
-              companion(paramNames)(
-                pt => registeringType(pt, paramReader.readParamTypes[PInfo](end)),
+              val (paramNames, mods) = nameReader.readParamNamesAndMods(end)
+              companionOp(mods)(paramNames.map(nameMap))(
+                pt => registeringType(pt, paramReader.readParamTypes[PInfo](paramNames.length)),
```

Comment:
I was potentially concerned about traversing the paramNames to get a length versus comparing to an end address, but I guess it is insignificant compared to the logic to traverse types

Reply:
Yes, indeed. Generally, we assume that `length` is free since most lists are small.